src/components/commerce/CommerceCourtReservationDetails.js

```diff
@@ -1,24 +1,153 @@
+      return (
+        <Button
+          title="Cancelar Reserva"
+          onPress={() =>
+            this.setState({ optionsVisible: !this.state.optionsVisible })
+          }
+        />
+      );
+    }
+  };
+
+  renderReasonInput = () => {
+    return (
+      <View style={{ alignSelf: 'stretch' }}>
+        <CardSection style={{ padding: 20, paddingLeft: 10, paddingRight: 10 }}>
+          <Input
+            placeholder="Motivos de cancelación..."
+            multiline={true}
+            color="black"
+            onChangeText={value => {
+              this.props.onCourtReservationsListValueChange({
+                prop: 'cancelationReason',
+                value
+              });
+              this.setState({ error: '' });
+            }}
+            value={this.props.cancelationReason}
+            errorMessage={this.state.error}
+            onFocus={() => this.setState({ error: '' })}
+          />
+        </CardSection>
+        <Divider style={{ backgroundColor: 'grey' }} />
+      </View>
+    );
+  };
+
+  renderError = () => {
+    if (this.props.cancelationReason === '') {
+      this.setState({ error: 'Debe informar el motivo' });
+      return false;
+    } else {
+      this.setState({ error: '' });
+      return true;
+    }
+  };
+
+  onBackdropPress = () => {
+    this.setState({ optionsVisible: false });
+    this.setState({ error: '' });
+    this.props.onCourtReservationsListValueChange({
+      prop: 'cancelationReason',
+      value: ''
+    });
+  };
+
+  onConfirmDelete = (id, clientId) => {
+    if (this.renderError()) {
+      this.setState({ optionsVisible: false });
+      this.props.onCommerceCancelReservation({
+        commerceId: this.props.commerceId,
+        reservationId: id,
+        clientId,
+        cancelationReason: this.props.cancelationReason,
+        navigation: this.props.navigation
+      });
     }
+  };
+
+  render() {
+    const {
+      client,
+      court,
+      startDate,
+      endDate,
+      price,
+      light,
+      id,
+      clientId
+    } = this.state.reservation;
+    return (
+      <View>
+        <CourtReservationDetails
+          client={client}
+          court={court}
+          startDate={startDate}
+          endDate={endDate}
+          price={price}
+          light={light}
+          showPrice={true}
+        />
+        <CardSection>{this.renderCancelButton()}</CardSection>
+
+        <Menu
+          title="Informar el motivo de la cancelación"
+          onBackdropPress={() => this.onBackdropPress()}
+          isVisible={this.state.optionsVisible}
+        >
+          {this.renderReasonInput()}
+          <MenuItem
+            title="Confirmar Cancelación"
+            icon="md-checkmark"
+            loadingWithText={this.props.loading}
+            onPress={() => this.onConfirmDelete(id, clientId)}
+          />
+          <MenuItem
+            title="Cerrar"
+            icon="md-close"
+            onPress={() => this.onBackdropPress()}
```

Comment:
Podrías ponerle `this.onBackdropPress` directamente

Reply:
No anda si le sacas el parentesis

Reply:
Para sacar el paréntesis, en vez de `onPress={() => this.onBackdropPress()}` , pone: `onPress={this.onBackdropPress}`
Pero eso deberíamos definir nosotros de que forma lo vamos a dejar, porque en varios lados se podría cambiar

Reply:
claro, cuando no recibe parametros se pasa asi de una, sino si pones como esta ahora, se esta creando una nueva funcion anonima que adentro llama a la otra